src/index.js

```diff
@@ -107,6 +161,68 @@ function getValueDoppelganger(key, originalValue) {
 }
 
 /**
+ * This takes a list of CSS values and converts it to an array
+ * @param {String} value - something like `1px`, `1px 2em`, or `3pt rgb(150, 230, 550) 40px calc(100% - 5px)`
+ * @return {Array} the split values (for example: `['3pt', 'rgb(150, 230, 550)', '40px', 'calc(100% - 5px)']`)
+ */
+function getValuesAsList(value) {
+  return value
+    .replace(/ +/g, ' ') // remove all extraneous spaces
+    .split(' ')
+    .map(i => i.trim()) // get rid of extra space before/after each item
+    .filter(Boolean) // get rid of empty strings
+     // join items which are within parenthese
+     // luckily `calc (100% - 5px)` is invalid syntax and it must be `calc(100% - 5px)`, otherwise this would be even more complex
+    .reduce(({list, state}, item) => {
+      if (includes(item, '(')) {
+        state.withinParens = true
+        list.push(item)
+      } else if (state.withinParens) {
+        if (includes(item, ')')) {
+          state.withinParens = false
+        }
+        list[list.length - 1] = `${list[list.length - 1]} ${item}`
+      } else {
+        list.push(item)
+      }
+      return {list, state}
+    }, {list: [], state: {withinParens: false}}).list
+}
+
+/**
+ * This is intended for properties that are `top right bottom left` and will switch them to `top left bottom right`
+ * @param {String} value - `1px 2px 3px 4px` for example, but also handles cases where there are too few/too many and
+ * simply returns the value in those cases
+ * @return {String} the result - `1px 4px 3px 2px` for example.
+ */
+function handleQuartetValues(value) {
+  const splitValues = getValuesAsList(value)
+  if (splitValues.length <= 3 || splitValues.length > 4) {
+    return value
+  }
+  const [top, right, bottom, left] = splitValues
+  return [top, left, bottom, right].join(' ')
+}
+
+/**
+ * Takes a percentage for background position and inverts it.
+ * @param {String} value - the original value (for example 77%)
+ * @return {String} the result (for example 23%)
+ */
+function calculateNewBackgroundPosition(value) {
+  const idx = value.indexOf('.')
+  if (idx === -1) {
+    value = `${100 - parseFloat(value)}%`
+  } else {
+    // Two off, one for the "%" at the end, one for the dot itself
```

Comment:
Haha, honestly, no idea, I yanked this logic from cssjanus. Should probably attribute this function to them...